src/api/mvc/tables/main.go

```diff
@@ -0,0 +1,306 @@
+
+	// table id must be provided
+	if tableId == "" {
+		fmt.Println("Missing table id")
+		return c.JSON(http.StatusBadRequest, errors.CreateSimpleBadRequest("Missing table id - please try again"))
+	}
+
+	// call repository
+	// - get the table by id
+	results, getTablesError := esRepo.GetTables(c, tableId, "")
+	if getTablesError != nil {
+		fmt.Printf("Failed to get tables with ID %s\n", tableId)
+		return c.JSON(http.StatusInternalServerError, errors.CreateSimpleInternalServerError("Something went wrong.. Please try again later!"))
+	}
+
+	// gather data from "hits"
+	docsHits := results["hits"].(map[string]interface{})["hits"]
+	if docsHits == nil {
+		fmt.Printf("No Tables with ID '%s' were deleted\n", tableId)
+		return c.JSON(http.StatusBadRequest, errors.CreateSimpleBadRequest(fmt.Sprintf("Table with ID %s not found", tableId)))
+	}
+
+	// obtain hits (expecting 1)
+	allDocHits := []map[string]interface{}{}
+	mapstructure.Decode(docsHits, &allDocHits)
+
+	// grab _source for each hit
+	allSources := make([]interface{}, 0)
+	// var allSources []indexes.Variant
+
+	for _, r := range allDocHits {
+		source := r["_source"]
+		byteSlice, _ := json.Marshal(source)
+
+		// cast map[string]interface{} to table
+		var resultingTable indexes.Table
+		if err := json.Unmarshal(byteSlice, &resultingTable); err != nil {
+			fmt.Println("failed to unmarshal:", err)
+		}
+
+		// accumulate structs
+		allSources = append(allSources, resultingTable)
+	}
+
+	if len(allSources) == 0 {
+		fmt.Printf("Failed to get table summary with ID '%s'\n", tableId)
+		return c.JSON(http.StatusBadRequest, errors.CreateSimpleBadRequest(fmt.Sprintf("Failed to get table summary with ID %s", tableId)))
+	}
+
+	// obtain table id from the one expected hit
+	// and search for variants associated with it
+
+	totalVariantsCount := 0.0
+
+	docs, countError := esRepo.CountDocumentsContainerVariantOrSampleIdInPositionRange(cfg, es,
+		chromosome, lowerBound, upperBound,
+		"", "", // note : both variantId and sampleId are deliberately set to ""
+		reference, alternative, genotype, assemblyId, tableId)
+	if countError != nil {
+		fmt.Printf("Failed to count variants with table ID %s\n", tableId)
+		return c.JSON(http.StatusInternalServerError, errors.CreateSimpleInternalServerError("Something went wrong.. Please try again later!"))
+	}
+
+	totalVariantsCount = docs["count"].(float64)
+
+	// obtain number of samples associated with this tableId
+	resultingBuckets, bucketsError := esRepo.GetVariantsBucketsByKeywordAndTableId(cfg, es, "sample.id.keyword", tableId)
+	if bucketsError != nil {
+		fmt.Println(resultingBuckets)
+	}
+
+	// retrieve aggregations.items.buckets
+	// and count number of samples
+	bucketsMapped := []interface{}{}
+	if aggs, ok := resultingBuckets["aggregations"]; ok {
+		aggsMapped := aggs.(map[string]interface{})
+
+		if items, ok := aggsMapped["items"]; ok {
+			itemsMapped := items.(map[string]interface{})
+
+			if buckets := itemsMapped["buckets"]; ok {
```

Comment:
Not sure about this, but it looks like the value for `ok` comes from the previous block, not the buckets assignment

Reply:
good catch!